src/MemStore.cc

```diff
@@ -360,9 +360,7 @@ MemStore::updateHeadersOrThrow(Ipc::StoreMapUpdate &update)
     // our +/- hdr_sz math below does not work if the chains differ [in size]
     Must(update.stale.anchor->basics.swap_file_sz == update.fresh.anchor->basics.swap_file_sz);
 
-    const HttpReply *rawReply = update.entry->getReply();
-    Must(rawReply);
-    const HttpReply &reply = *rawReply;
+    const auto &reply = update.entry->mem().baseReply();
```

Comment:
I assume we need freshestReply() instead, since baseReply() contains unmodified headers:

```suggestion
    const auto &reply = update.entry->mem().freshestReply();
```

This is why my Daft test (which appended headers on 304) did not work.

Reply:
I think we need both -- staleHdrSz should get information from the base reply while packHeadersUsingSlowPacker() should use the freshest one. I will fix.

Reply:
Fixed in 29a0bed.